vaadin-cdi/src/test/java/com/vaadin/cdi/SessionContextTest.java

```diff
@@ -31,9 +32,34 @@ public void testUIsAccessSameSession() throws Exception {
     }
 
     @Test
-    public void testSessionCloseDestroysSessionContext() throws Exception {
+    public void testVaadinSessionCloseDestroysSessionContext() throws Exception {
         Assert.assertEquals(0, getCount(SessionUI.SessionScopedBean.DESTROY_COUNT));
         clickAndWait(SessionUI.INVALIDATEBTN_ID);
         Assert.assertEquals(1, getCount(SessionUI.SessionScopedBean.DESTROY_COUNT));
     }
+
+    @Test
+    public void testHttpSessionCloseDestroysSessionContext() throws Exception {
+        Assert.assertEquals(0, getCount(SessionUI.SessionScopedBean.DESTROY_COUNT));
+        clickAndWait(SessionUI.HTTP_INVALIDATEBTN_ID);
+        Assert.assertEquals(1, getCount(SessionUI.SessionScopedBean.DESTROY_COUNT));
+    }
+
+    @Test
+    @Ignore
+    //ignored because it's slow, and expiration should be same as session close
+    public void testHttpSessionExpirationDestroysSessionContext() throws Exception {
```

Comment:
Revise whether
    remove this test,
    move to a slow test group
    speed it up somehow
    or just leave ignored